tokio-timer/src/delay_queue.rs

```diff
@@ -496,15 +496,22 @@ impl<T> DelayQueue<T> {
 
         // Normalize the deadline. Values cannot be set to expire in the past.
         let when = self.normalize_deadline(when);
-        let old = self.start + Duration::from_millis(self.slab[key.index].when);
 
+        // This is needed only for the debug assertion inside the if-let, but
+        // calculating it there causes the assertion to fail for some reason.
+        #[cfg(debug_assertions)]
```

Comment:
Is the cfg necessary to silence warnings?

Reply:
LLVM should be pretty good at this type of dead code removal, FWIW.

Reply:
This is true, I typically just don't like having to _rely_ on optimizer behaviour.